xbmc/windows/GUIMediaWindow.cpp

```diff
@@ -823,8 +822,7 @@ bool CGUIMediaWindow::Update(const std::string &strDirectory, bool updateFilterP
   m_vecItems->FillInDefaultIcons();
 
   // remember the original (untouched) list of items (for filtering etc)
-  m_unfilteredItems->SetPath(m_vecItems->GetPath()); // use the original path - it'll likely be relied on for other things later.
-  m_unfilteredItems->Append(*m_vecItems);
+  m_unfilteredItems->Assign(*m_vecItems);
```

Comment:
using `Assign()` here will clear the file items before and copy over properties like path etc.